Sources/Dependencies/DependencyValues/MainQueue.swift

```diff
@@ -53,7 +54,7 @@
     }
 
     private enum MainQueueKey: DependencyKey {
-      static let liveValue = AnySchedulerOf<DispatchQueue>.main
+      static let liveValue = DispatchQueue.main.dependencies()
       static let testValue = AnySchedulerOf<DispatchQueue>
         .unimplemented(#"@Dependency(\.mainQueue)"#)
```

Comment:
I was worried that users would explicitly need to use `dependencies()` in the test value, but then I realized that test schedulers already forward dependencies since they invoke the closure immediately (inheriting task locals). Though I'm not sure if this applies to the delay/interval APIs; dunno if we need to add explicit tests for those.